internal/util/util.go

```diff
@@ -329,19 +330,33 @@ func Sample(m proto.Message, strategy *SamplingStrategy) proto.Message {
 		// Note, map[string]proto.Message is treated as protoreflect.MessageKind,
 		// So here need to check field and list first.
 		if fd.IsList() {
-			// Sampmle list.
+			// Sample list.
 			oldList := value.List()
-			inc := int(1 / strat.Ratio)
+			length := oldList.Len()
 			var newList protoreflect.List
-			for i := 0; i < oldList.Len(); i += inc {
-				newList.Append(oldList.Get(i))
+
+			maxSample := strat.MaxSample
+			if strat.MaxSample == -1 || strat.MaxSample > length {
+				maxSample = length
+			}
+			inc := 1
+			if length > maxSample {
+				inc = int(math.Ceil(float64(length) / float64(maxSample)))
+			}
+			// Get the latest data first
+			for i := 0; i < maxSample; i++ {
```

Comment:
hm, the tracked keys won't stay stable this way, so the diffs will be harder to inspect even though the files are smaller (e.g. we won't know if values change, as data gets added which happens for covid).

Reply:
Yea, if we sample the time series, the stability would indeed be an issue. It's possible the covid addition would cause entire diff of the series.

Or we could go back to the old approach that only samples the place but full data for that place. It's much more stable.

Reply:
We can probably try the current approach and adjust the SampleStrategy to place only if that's better?

Reply:
Chatted offline, and we'll try place sampling only and see how that works out.
Thanks Bo, and sorry for the back and forth.

Reply:
No worry. Switched back to sample the places only